src/Cli/dotnet/commands/dotnet-workload/install/IWorkloadManifestUpdater.cs

```diff
@@ -8,20 +8,24 @@ namespace Microsoft.DotNet.Workloads.Workload.Install
 {
     internal interface IWorkloadManifestUpdater
     {
-        Task UpdateAdvertisingManifestsAsync(bool includePreviews, DirectoryPath? offlineCache = null);
+        Task UpdateAdvertisingManifestsAsync(bool includePreviews, DirectoryPath? offlineCache = null, IEnumerable<WorkloadManifestInfo> manifests = null);
```

Comment:
I wanted to store it in a workload history record, but the UpdateAdvertisingManifestsAsync method call happens without having calculated that particular argument, so I could either pass in the WorkloadHistoryRecorder and update its argument within the UpdateAdvertisingManifestsAsync method, or I could calculate the installed manifests early. I chose the latter approach, but once it's calculated, there's no reason not to reuse it rather than calculating it again, so that's why I added an argument.